### PR TITLE
Add gradient, Hessian, and orbital coeffs scratch file parsers to `pymatgen.io.qchem.outputs`

### PR DESCRIPTION
**Pull request by Andrew-S-Rosen**
## Summary

This PR adds function to `pymatgen.io.qchem.outputs` to parse (scratch) gradient, Hessian, and previous orbital coefficient files made by Q-Chem.

**Note**: 99% of the credit for this goes to @samblau. I am simply moving it into pymatgen where it belongs.

## Todos

I need to add tests. There are also some things that are definitely better done with numpy vectorization, but it's not a huge issue.

## Checklist

- [X] Google format doc strings added. Check with `ruff`.
- [X] Type annotations included. Check with `mypy`.
- [x] Tests added for new features/fixes.
- [] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

Tip: Install `pre-commit` hooks to auto-check types and linting before every commit:

```sh
pip install -U pre-commit
pre-commit install
```

### CODE REVIEW

**Comment by Andrew-S-Rosen** [comment]
@janosh: This should be ready for review. As I mentioned above, this is code mostly written my @samblau --- I added some simple tests and cleaned up a few things. Longer term, it would probably make sense to create a new `QCScratch(MSONable)` class that parses scratch files in a directory. However, I have not implemented that here (unless you feel it necessary).

File: tests/files/qchem/53.0
Comment (janosh) [inline tests/files/qchem/53.0:1]: Given the file size probably worth compressing these test files, esp. this one?

**Review by janosh — CHANGES_REQUESTED** [review]
Thanks @Andrew-S-Rosen, looks great! Just a minor suggestion re test file compression.

File: tests/io/qchem/test_outputs.py [hunk 1]
```diff
@@ -492,6 +501,37 @@ def test_nbo_3_c(self):
         assert perturb_ene[0]["perturbation energy"][3209] == 3.94
 
 
+def test_gradient(tmpdir):
```
Comment (janosh) [inline tests/io/qchem/test_outputs.py:504]: I spoke to a `pytest` developer once who said `tmpdir` is slated for eventual deprecation so better to use `tmp_path` here.

Comment (Andrew-S-Rosen) [reply tests/io/qchem/test_outputs.py]: @janosh: Thanks! Didn't know about that. I just changed them out. Resolving this.

Comment (Andrew-S-Rosen) [reply tests/io/qchem/test_outputs.py]: Ew, that's gross. I forgot since it's `Path` now, it doesn't have that. Changed. Hopefully this works.

**Comment by Andrew-S-Rosen** [comment]
@janosh: All good now!

**Review by janosh — APPROVED** [review]
Thanks @Andrew-S-Rosen! 👍

**Pull request merged**